libsoftwarecontainer/src/gateway/network/networkgateway.cpp

```diff
@@ -85,7 +82,9 @@ bool NetworkGateway::activateGateway()
         return false;
     }
 
-    if (isError(generateIP())) {
+
+    if (false == m_functions.generateIP(m_netmask, m_gateway, m_containerID, m_ip)) {
```

Comment:
actuall calling it with ! operator is not good when it comes to readability cause the called function was long with mant arguments. 

But anyways this is changed  as you wanted with exceptions.